src/main/java/software/amazon/awssdk/s3accessgrants/plugin/S3AccessGrantsIdentityProvider.java

```diff
@@ -72,6 +74,9 @@ public class S3AccessGrantsIdentityProvider implements IdentityProvider<AwsCrede
 
     private String CONTACT_TEAM_MESSAGE_TEMPLATE = "An internal exception has occurred. Valid %s was not passed to the %s. Please contact S3 access grants plugin team!";
 
+    ClientOverrideConfiguration.Builder overrideConfig = ClientOverrideConfiguration.builder()
+                    .putAdvancedOption(SdkAdvancedClientOption.USER_AGENT_PREFIX, "aws-s3-accessgrants-java-sdk-v2-plugin");
+
```

Comment:
Can we also have an option to append the user provided user-agent string here. Reg: https://github.com/aws/aws-s3-accessgrants-plugin-java-v2/issues/132